### PR TITLE
Add multiple port support

### PR DESCRIPTION
**Pull request by punmechanic**
# Context

As part of the [OAuth2 authorization code flow][0] keyconjurer conducts to retrieve an access token for the user, keyconjurer currently listens on the address `0.0.0.0:57468` for inbound HTTP requests.

As per [RFC 8252][rfc8252], IdP MUST allow for ephemeral port support:

>    An example redirect using the IPv6 loopback interface with a randomly
>    assigned port:
> 
>      http://[::1]:61023/oauth2redirect/example-provider
> 
>    The authorization server MUST allow any port to be specified at the
>    time of the request for loopback IP redirect URIs, to accommodate
>    clients that obtain an available ephemeral port from the operating
>    system at the time of the request.

Unfortunately, Okta does not support this; It is not possible to specify any port using a wildcard; all ports must be explicitly specified in the OAuth2 client configuration.

# The problem

At Riot, we have some users who are using Windows in particular that are finding that Windows is assigning the port `57468` to other processes. It is possible that Windows could assign that port number as we are not making any reservations or modifications to the Windows registry - Doing so would likely involve using an installer or a great deal of Windows-specific code which there isn't the appetite to author right now; if we were to take that step it would be easier to simply make a custom URL scheme and bypass ports entirely (similar to `spotify://`).

This PR attempts to resolve this by allowing KeyConjurer to try a number of different ports (which each must be explicitly passlisted by the IdP) before giving up and reporting an error to the user.

[0]: https://oauth.net/2/grant-types/authorization-code/
[rfc8252]: https://datatracker.ietf.org/doc/html/rfc8252#section-7.3

### CODE REVIEW

File: cli/consts.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ var (
 	Version        = "TBD"
 	BuildTimestamp = "BuildTimestamp is not set"
 	DownloadURL    = "URL not set yet"
+	CallbackPorts  = []string{"57468", "58888", "59999", "60000"}
```
Comment (artem-smotrakov) [inline cli/consts.go:11]: Do you know how many ports Okta allows? I am wondering if we can add more if that makes sense.

File: cli/get.go [hunk 2]
```diff
@@ -39,11 +39,13 @@ func init() {
...
-	getCmd.Flags().String(FlagAWSCLIPath, "~/.aws/", "Path for directory used by the aws-cli tool. Default is \"~/.aws\".")
 	getCmd.Flags().String(FlagTencentCLIPath, "~/.tencent/", "Path for directory used by the tencent-cli tool. Default is \"~/.tencent\".")
 	getCmd.Flags().String(FlagCloudType, "aws", "Choose a cloud vendor. Default is aws. Can choose aws or tencent")
 	getCmd.Flags().Bool(FlagBypassCache, false, "Do not check the cache for accounts and send the application ID as-is to Okta. This is useful if you have an ID you know is an Okta application ID and it is not stored in your local account cache.")
 	getCmd.Flags().Bool(FlagLogin, false, "Login to Okta before running the command")
+	getCmd.Flags().String(FlagAWSCLIPath, "~/.aws/", "Path for directory used by the aws CLI")
+	getCmd.Flags().BoolP(FlagURLOnly, "u", false, "Print only the URL to visit rather than a user-friendly message")
+	getCmd.Flags().BoolP(FlagNoBrowser, "b", false, "Do not open a browser window, printing the URL instead")
```
Comment (artem-smotrakov) [inline cli/get.go:48]: What would be the use cases for these options? Is it supposed to be used by users or dev for debugging purposes? Can you give a few examples?

**Review by artem-smotrakov — APPROVED** [review]
Looks good to me. Left a few questions.

File: cli/consts.go (hunk 1 above)
Comment (punmechanic) [reply cli/consts.go:11]: I'm not sure but I want to try to increase this to 16 ports that are non-contiguous from the 40 to 50k range

File: cli/get.go (hunk 2 above)
Comment (punmechanic) [reply cli/get.go:48]: These options are mirroring the options in `keyconjurer login` and are meant to be used with `keyconjurer get --login account-name`. 

The main use-case for these is that if you specify the `--login` flag, you can control the behavior of keyconjurer, where keyconjurer either:

* opens a browser (default)
* displays a url only (not the most user intuitive)
* displays a human-friendly message on opening a browser with a given url (default if `-b` specified)

**Pull request merged**